api/app.go

```diff
@@ -41,3 +42,44 @@ func (h apiHandler) SaveConfiguration(w http.ResponseWriter, r *http.Request) {
 
 	h.NoContent(w)
 }
+
+func (h apiHandler) PerformMaintenance(w http.ResponseWriter, r *http.Request) {
+	var req AppMaintenanceRequest
+	if err := readJSONFromRequest(r, &req); err != nil {
+		h.Error(w, r, http.StatusBadRequest, err)
+		return
+	}
+
+	switch req.Op {
+	case OptimizeImages:
+		allStates := make([]models.RecipeState, 2)
+		allStates = append(allStates, models.Active)
+		allStates = append(allStates, models.Archived)
+		allFilter := models.SearchFilter{
+			Query:        "",
+			Fields:       make([]models.SearchField, 0),
+			Tags:         make([]string, 0),
+			WithPictures: nil,
+			States:       allStates,
+			SortBy:       models.SortById,
+			SortDir:      models.Asc,
+		}
+		// TODO: Paging?
+		allRecipes, _, err := h.db.Recipes().Find(&allFilter, 1, 1000000)
+		if err != nil {
+			h.Error(w, r, http.StatusInternalServerError, err)
+			return
+		}
+		for _, recipe := range *allRecipes {
+			err = upload.OptimizeImages(h.upl, *recipe.Id)
+			if err != nil {
+				// TODO: Log and continue?
+				h.Error(w, r, http.StatusInternalServerError, err)
+				return
+			}
+		}
+	default:
+	}
+
+	h.NoContent(w)
```

Comment:
Accepted? Should this route return before completion?

Reply:
Perhaps some of the logic should be on the client side instead? Loop over all recipes and images client size, and send an optimize command from there?